src/routes/ProtectedDashboardRoute/ProtectedDashboardRoute.tsx

```diff
@@ -0,0 +1,53 @@
+import { useAppDispatch, useAppSelector } from "@/app/hooks"
+import { logoutUser } from "@/features/auth/authSlice"
+import getNewAccessToken from "@/libs/axios/getNewAccessToken"
+import {
+  getAccessTokenFromLocalStorage,
+  getRefreshTokenFromLocalStorage,
+} from "@/utils/localStorage"
+import { jwtDecode } from "jwt-decode"
+import { ReactElement, useEffect } from "react"
+import { Navigate, useLocation } from "react-router-dom"
+
+interface ProtectedDashboardRouteProps {
+  children: ReactElement
+}
+
+const ProtectedDashboardRoute = ({
+  children,
+}: ProtectedDashboardRouteProps) => {
+  const { isAuthenticated } = useAppSelector((store) => store.auth)
+  const dispatch = useAppDispatch()
+  const location = useLocation()
+  const accessToken = getAccessTokenFromLocalStorage()
+  const refreshToken = getRefreshTokenFromLocalStorage()
+
+  const checkAuth = () => {
```

Comment:
Wrap this function in useCallback since it will cause the useEffect to re-render